City-of-Helsinki/drupal-helfi-platform-config
# UHF-9497 Word vs. CKEditor5

# [UHF-9497](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9497)


## What was done


* Removed obsolete attributes from full and minimal filter formats.
* Added an update hook to convert old link button data-attributes to follow new HDS data-attributes.
* Removed class attributes from anchor tags in filter formats.
* Removed class, aria-hidden and role attributes from span elements in filter formats.
* Removed span from editor configurations.
* Added CKEditor5 paste filter module and configured settings for it.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-9497 drupal/hdbt:dev-UHF-9497 -W`
* Run `make drush-updb drush-cr`

## How to test


This PR / Update will affect all `<a>` tags in long text fields (CKEeditor fields). In hel.fi there are more than 2500 links which still have old data attributes and old classes. This update will go through each of these fields and convert the following data attributes: 
```
data-selected-icon --> data-hds-icon-start
data-design="hds-button hds-button--primary" --> data-hds-component="button"
data-design="hds-button hds-button--secondary" --> data-hds-component="button" data-hds-variant="secondary"
data-design="hds-button hds-button--supplementary" --> data-hds-component="button" data-hds-variant="supplementary"
```
Following will be removed: 
```
data-design="link"
data-link-text="*"
class="hds-button hds-button--primary"
class="hds-button hds-button--secondary"
class="hds-button hds-button--supplementary"
```

As going through all of the changed elements/attributes is pointless, I've gathered few links from SOTE which should've been updated after running the updb. If you're interested of a complete list, contact @Arkkimaagi.

After your environment is up and running, open these links:
- https://www.hel.fi/fi/sosiaali-ja-terveyspalvelut/vammaispalvelut/tyo-ja-paivatoiminta/tyohonvalmennus
   - There is a "Katso video työhönvalmennuksesta" supplementary button
- https://www.hel.fi/fi/sosiaali-ja-terveyspalvelut/senioripalvelut/seniorikeskukset-etusivu/kivela/osastot   
   - There are several "Siirry yhteystietoihin sivulle numerot.hel.fi" buttons
- https://www.hel.fi/fi/sosiaali-ja-terveyspalvelut/senioripalvelut/kotihoito/kuntouttava-arviointi#kuntouttavan-arviointiyksikon-tiimien-yhteystiedot
   - There are few phone number buttons
- https://www.hel.fi/sv/social-och-halsovardstjanster/tjanster-for-barn-och-familjer/stod-for-barn-unga-och-familjer/stod-for-foraldraskapet-och-familjens-vardag/familjesocialarbete#sa-har-kontaktar-du-oss-i-maisa
   - There is a "Gå till Maisa" link with an extra icon  

Here are the same links to local environment:
- https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/vammaispalvelut/tyo-ja-paivatoiminta/tyohonvalmennus
- https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/senioripalvelut/seniorikeskukset-etusivu/kivela/osastot
- https://helfi-sote.docker.so/fi/sosiaali-ja-terveyspalvelut/senioripalvelut/kotihoito/kuntouttava-arviointi#kuntouttavan-arviointiyksikon-tiimien-yhteystiedot
- https://helfi-sote.docker.so/sv/social-och-halsovardstjanster/tjanster-for-barn-och-familjer/stod-for-barn-unga-och-familjer/stod-for-foraldraskapet-och-familjens-vardag/familjesocialarbete#sa-har-kontaktar-du-oss-i-maisa

* [x] Go through the links mentioned above and check that the markup has been much more clear now. See source code example below. The upper half is a screenshot of local environment after update hook run and bottom half is from production.
![image](https://github.com/City-of-Helsinki/drupal-helfi-platform-config/assets/1712902/35dfe5d3-0da6-4b68-996e-1e063d50d8cd)
* [x] Create a banner and hearings paragraphs and check that the link buttons still look the same as before. [Their templates have been updated to use data-variables.](https://github.com/City-of-Helsinki/drupal-hdbt/pull/909/files#diff-c9e96b9e3447a59c3b447af7268809af13c11e376951976b9a569f24a40a60cbR22)
* [x] Go to [UHF-9497](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9497) and copy the word document from the issue attachments
   * [x] Paste the word document contents to Text paragraph and check that it's not that messy as it used to be - you can verify it by pasting it to a drupal instance which has a clean install

## Other PRs


* https://github.com/City-of-Helsinki/drupal-hdbt/pull/909
* https://github.com/City-of-Helsinki/drupal-helfi-strategia/pull/588
* https://github.com/City-of-Helsinki/drupal-helfi-asuminen/pull/362
* https://github.com/City-of-Helsinki/drupal-helfi-kuva/pull/299
* https://github.com/City-of-Helsinki/drupal-helfi-kasvatus-koulutus/pull/577
* https://github.com/City-of-Helsinki/drupal-helfi-rekry/pull/413
* https://github.com/City-of-Helsinki/drupal-helfi-sote/pull/769
* https://github.com/City-of-Helsinki/drupal-helfi-kymp/pull/803
* https://github.com/City-of-Helsinki/drupal-helfi-etusivu/pull/533
* https://github.com/City-of-Helsinki/drupal-helfi-tyo-yrittaminen/pull/350


[UHF-9497]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9497?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
